### PR TITLE
[1.17] Add support for host_rewrite_header

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Backport of https://github.com/solo-io/gloo/pull/9608

Adds support for the `host_rewrite_header` route option.
This configures envoy to rewrite the host header with the value of another header based on the value provided
Ref: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-host-rewrite-header

This also fixes a bug where gloo did not reject invalid envoy header keys
`vs.yaml`
```
        headerManipulation:
          requestHeadersToAdd:
          - header:
              key: "head(er)"
              value: random
```

Previously
```
$ kg apply -f vs.yaml 
virtualservice.gateway.solo.io/default created

$ curl localhost:8080/get
invalid header name: head(er)
```

Now
```
$ kg apply -f vs.yaml 

	* Validating *v1.VirtualService failed: validating *v1.VirtualService name:"default" namespace:"gloo-system": 1 error occurred:
	* failed gloo validation resource reports: 2 errors occurred:
	* invalid resource gloo-system.gateway-proxy
	* Route Error: ProcessingError. Reason: *headers.plugin: 'head(er)' is an invalid HTTP header key. Route Name: gloo-system_default-route-0-matcher-0; Route Error: ProcessingError. Reason: *headers.plugin: 'head(er)' is an invalid HTTP header key. Route Name: gloo-system_default-route-0-matcher-0
```

## API changes
- Adds the `host_rewrite_header` to the route options

## Code changes
- Adds support for the `host_rewrite_header` option in the basic route plugin

## CI changes
- Disables ingress tests on the v1.17.x branch

## Docs changes
- N/A

# Context
https://github.com/solo-io/gloo/issues/9579

## Interesting decisions
 N/A

## Testing steps
Unit tests

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/9622

### CODE REVIEW

File: .github/workflows/regression-tests.yaml [hunk 1]
```diff
@@ -47,9 +47,9 @@ jobs:
       fail-fast: false
       matrix:
         # upgrade tests are run on LTS but not on main branch, for main they are run nightly
-        # ingress will be deprecated from 1.17. Ref: https://solo-io-corp.slack.com/archives/G01EERAK3KJ/p1716389614777799
```
Comment (sheidkamp) [inline .github/workflows/regression-tests.yaml:50]: Were these changes intended to be part of this PR?

**Review by sheidkamp — COMMENTED** [review]
Question about the regression test change

File: .github/workflows/regression-tests.yaml (hunk 1 above)
Comment (davidjumani) [reply .github/workflows/regression-tests.yaml:50]: They should have been made when the 1.17 branch was cut. Added them here

Comment (sheidkamp) [reply .github/workflows/regression-tests.yaml:50]: NIT: can you note that in the CI section of the PR description?

**Review by sheidkamp — APPROVED** [review]

File: .github/workflows/regression-tests.yaml (hunk 1 above)
Comment (davidjumani) [reply .github/workflows/regression-tests.yaml:50]: Added

**Review by nfuden — APPROVED** [review]

**Comment by davidjumani** [comment]
kick bulldozer
